freqtrade/strategy/strategy_helper.py

```diff
@@ -83,3 +85,56 @@ def stoploss_from_open(open_relative_stop: float, current_profit: float) -> floa
 
     # negative stoploss values indicate the requested stop price is higher than the current price
     return max(stoploss, 0.0)
+
+
+def get_custom_dataframe(self, pair: str):
+    dataframe, _ = self.dp.get_analyzed_dataframe(pair, self.timeframe)
+    return dataframe
```

Comment:
If it's not in a class, there won't be any `self`.

As such - you'll need to remove all references to "self" - and add the eventually necessary parameters to the function head.

The above is valid for all methods below as well.


For that particular method:
with that - this method will do nothing - and can (and should) be inlined into the strategy - so at least this particular function can for sure be removed.